treeherder/etl/management/commands/ingest.py

```diff
@@ -425,17 +432,33 @@ def handle(self, *args, **options):
             # get reference to repo and ingest this particular revision for this project
             project = options["project"]
             commit = options["commit"]
+
+            if not options['last_n_pushes'] and not commit:
+                raise CommandError('must specify --last_n_pushes or a positional commit argument')
+
+            # get reference to repo
             repo = Repository.objects.get(name=project, active_status="active")
-            pushlog_url = "%s/json-pushes/?full=1&version=2" % repo.url
-            process = HgPushlogProcess()
-            process.run(pushlog_url, project, changeset=commit, last_push_id=None)
 
-            if options["ingest_all_tasks"]:
+            if options['last_n_pushes']:
+                last_push_id = last_push_id_from_server(repo)
+                fetch_push_id = max(1, last_push_id - options['last_n_pushes'])
+                logger.info('last server push id: %d; fetching push %d and newer',
+                            last_push_id, fetch_push_id)
+            elif options["ingest_all_tasks"]:
                 gecko_decision_task = get_decision_task_id(project, commit, repo.tc_root_url)
                 logger.info("## START ##")
                 loop.run_until_complete(processTasks(gecko_decision_task, repo.tc_root_url))
                 logger.info("## END ##")
             else:
                 logger.info(
                     "You can ingest all tasks for a push with -a/--ingest-all-tasks."
+                    "or last n pushes with --last-n-pushes"
```

Comment:
No need to mention this here.
The only reason I mention --ingest-all-tasks is because it is computationally expensive while last N pushes is not expensive.